homeassistant/components/dsmr_reader/config_flow.py

```diff
@@ -0,0 +1,46 @@
+"""Config flow to configure DSMR Reader."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.components.mqtt import MqttServiceInfo
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class DsmrReaderFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle DSMR Reader config flow."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import from configuration.yaml and create config entry."""
+        # There's no configuration supported for this integration, so we need to hardcode the user input here
+        return await self.async_step_user(user_input={})
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle user step."""
+        if self._async_current_entries():
+            return self.async_show_form(
```

Comment:
I think it would be better with aborting the flow on this error

Reply:
That makes more sense, indeed. Adjusted.

Reply:
Is this actually possible since `mqtt` is in `dependencies`?

Reply:
I guess it is if they have no mqtt config and they try to create this manually.